### PR TITLE
fix: align view after jumplist_picker

### PR DESCRIPTION
**Pull request by shnarazk**
Since `jumplist_picker` doesn't update screen, we need to do one action to get the correct view.
So I added 'align_view' to the callback like `jump_backward`.

### CODE REVIEW

**Comment by shnarazk** [comment]
OK. I added `View::offset_coords_to_in_view_for_search` that calculates offsets for two `scrolloff`s, based on `offset_coords_to_in_view`. And all of `search_impl` and 3 jump commands were revised to use another new method `View::place_cursor_in_view`, which tries to fix the view. 
Now, this PR is about 'aligning view after jump commands'...

**Comment by archseer** [comment]
Hmm this seems to have some other commits mixed in now?

**Comment by shnarazk** [comment]
@archseer 
Oh, sorry. I've just rebased and squashed. Please check this again.

File: helix-view/src/view.rs [hunk 1]
```diff
@@ -201,6 +202,53 @@ impl View {
...
+                // scroll right
+                self.offset.col + col - (last_col.saturating_sub(scrolloff))
+            } else if col < self.offset.col + scrolloff {
+                // scroll left
+                col.saturating_sub(scrolloff)
+            } else {
+                self.offset.col
+            };
+            (row, col)
+        };
```
Comment (the-mikedavis) [inline helix-view/src/view.rs:248]: It looks like this code ends up being mostly duplicated between this and the above function. Can we factor out the shared parts into a function they both call?

**Review by the-mikedavis — COMMENTED** [review]
This behavior seems like it might be a good default. Should we just use this behavior for all instances of `view.ensure_cursor_in_view(doc, scrolloff)`?

File: helix-view/src/view.rs (hunk 1 above)
Comment (shnarazk) [reply helix-view/src/view.rs]: @the-mikedavis 
Hmmm, `offset_coords_to_in_view` is a simplified version of this, more generic function. So I revised `offset_coords_in_view_center`to take an extra argument, and `offset_coords_to_in_view` as a wrapper for it.
Also, I renamed new methods and changed their args.

**Comment by the-mikedavis** [comment]
It looks like there's a regression with the behavior of scrolloff - if you hold `j` on a document longer than the screen, helix behaves as if it's configured to use a scrolloff of 0

**Comment by shnarazk** [comment]
@the-mikedavis 
My bad. The else block of the last if statement was revised to use `scrolloff`.

**Review by the-mikedavis — APPROVED** [review]

**Pull request merged**